src/extractor/guidance/intersection_handler.cpp

```diff
@@ -393,15 +393,18 @@ std::size_t IntersectionHandler::findObviousTurn(const EdgeID via_edge,
 
     const EdgeData &in_way = node_based_graph.GetEdgeData(via_edge);
 
+    const auto IsContinueRoad = [&](const EdgeData &way_data) {
+        return !util::guidance::requiresNameAnnounced(
+            in_way.name_id, way_data.name_id, name_table, street_name_suffix_table);
+    };
```

Comment:
I'm doing the same over in my Sliproad Refactoring

https://github.com/Project-OSRM/osrm-backend/blob/eae226a9b1685ceba39589856afb24b266fbed9c/src/extractor/guidance/sliproad_handler.cpp#L589-L606

Maybe we should put it into the IntersectionHandler base class then we can both use it.